src/main/scala/mesosphere/marathon/state/Volume.scala

```diff
@@ -112,8 +112,12 @@ case class PersistentVolume(
 
 object PersistentVolume {
   import org.apache.mesos.Protos.Volume.Mode
+
+  val NoSlashesPattern = """^[^/]+$""".r
+
   implicit val validPersistentVolume = validator[PersistentVolume] { vol =>
-    vol.containerPath is notEmpty
+    vol.containerPath is notOneOf(".", "..")
+    vol.containerPath should matchRegexFully(NoSlashesPattern)
```

Comment:
I understand the intention of the `notOneOf`, but I'm not sure I like splitting this up into matching against those _plus_ matching against the pattern. Why not only use a pattern? The `notOneOf` could still be used elsewhere so I'd definitely keep it as a enhancement (or create a PR against accord).

In order to have an _always meaningful_ validation error message, you could use the `isTrue` function with a constraint description.